### PR TITLE
update Libseccomp to tagged release (0.9.1)

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: updates github.com/seccomp/libseccomp-golang to 0.9.1 (the latest tagged release), in order to do this also works around gazelle's lack of `pkg-config` support (which will not be added https://github.com/bazelbuild/bazel-gazelle/pull/556#issuecomment-504397812).

**Which issue(s) this PR fixes**:

Part of https://github.com/kubernetes/kubernetes/issues/79234

See also https://github.com/kubernetes/kubernetes/pull/78187

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/update-bazel.sh
Comment (BenTheElder) [inline hack/update-bazel.sh]: This is the same think we had before, modulo formatting. So far experimentation suggests that `-lseccomp` is the only link flag `pkg-config` would give us if we were using `go build` / CGO so I think this should be fine.

Without this trick here we lose the link option in the bazel build.

I don't love this, but I like it a lot more than using ancient untagged versions of libraries, and more than dealing with removing bazel or something...

It should be functionally equivalent to how we built before, just with this snippet here correcting the BUILD file manually.

Comment (cblecker) [reply hack/update-bazel.sh]: needs ret logic at EOL.

Comment (BenTheElder) [reply hack/update-bazel.sh]: thanks, I spent so much time futzing with the patching that I forgot this :(

Comment (BenTheElder) [reply hack/update-bazel.sh]: fixed

File: vendor/github.com/seccomp/libseccomp-golang/seccomp.go
Comment (odinuge) [inline vendor/github.com/seccomp/libseccomp-golang/seccomp.go]: This will still give a couple of lines with warnings that say `error`, when running update-bazel.sh.

```bash
> ./hack/update-bazel.sh                      
~/go/src/k8s.io/kubernetes/vendor ~/go/src/k8s.io/kubernetes
~/go/src/k8s.io/kubernetes
gazelle: /home/odin/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp.go: error reading go file: /home/odin/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp.go: pkg-config not supported: #cgo pkg-config: libseccomp
gazelle: /home/odin/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go: error reading go file: /home/odin/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go: pkg-config not supported: #cgo pkg-config: libseccomp
```

I think that it may look like errors for people without the insight into this.. Should we filter them out (hide them from the user), or should we just leave them for now? I am leaning towards filtering it out, since it annoys me quite a bit...

**Review by odinuge — APPROVED** [review]
Since `pkg-config` is kinda pain when using bazel (and similar utils), I think this is the best solution atm. The newest release of gazelle does not remove te clinkopts when running `gazelle fix`, so updating is another solution (but then we will have to look at the logic of deleting all `BUILD` files inside vendor when updating deps. I like keeping it explicit, so I like the this solution! :smile:

The new `runc` version in https://github.com/kubernetes/kubernetes/pull/78495 (that we hopefully can get into master quickly) depends on this version of libseccomp-golang, so getting it up to date is nice!

So other than the comment below, LGTM! :+1:

**Comment by BenTheElder** [comment]
> The newest release of gazelle does not remove te clinkopts when running gazelle fix, so updating is another solution (but then we will have to look at the logic of deleting all BUILD files inside vendor when updating deps. I like keeping it explicit, so I like the this solution! 😄

I actually looked at doing this and updated the `find .. -exec rm ...` logic to not touch this file, but `go mod vendor` also rewrites all of `vendor/` so that route is not super feasible unfortunately. I think this route is relatively fine though, my bigger concern is needing to hardcode the link options ourselves in divergence with the upstream library.

File: vendor/github.com/seccomp/libseccomp-golang/seccomp.go
Comment (BenTheElder) [reply vendor/github.com/seccomp/libseccomp-golang/seccomp.go]: I'm not sure what we should do here tbh. This is a bit annoying but the warnings are legitimate, it might be ok if we specifically filtered the warning for this package since we have a work around, but it probably would be a bad idea to generally hide them from all packages... 🤔

Comment (odinuge) [reply vendor/github.com/seccomp/libseccomp-golang/seccomp.go]: Yes, it feel a bit hacky, but since we already do some funky stuff with the ldflags, i guess filtering out this warning/error is reasonable atm. Doing something like this to explicitly filter out warnings from the two files:

```bash
$ gazelle fix "...." 2>&1 | grep -Ev "vendor/github.com/seccomp/libseccomp-golang/seccomp(_internal)?.go: pkg-config not supported: #cgo pkg-config: libseccomp"
```

The PR in libseccomp-golang introducting this, https://github.com/seccomp/libseccomp-golang/pull/8, only says _Not all distributions package libseccomp in the same way, but pkg-config
allows the same configuration to work on different distributions_, but we have never had any problems with this (the old one), so it looks like all distros use the same ldflag. Maybe we can ask them to switch back? I agree that `pkg-config` is nice when dealing with different operating systems, or with more advanced stuff that require a lot of ld- and libflags, but this code will only run on linux, and is quite simple with its single ldflag... :thinking:

Comment (BenTheElder) [reply vendor/github.com/seccomp/libseccomp-golang/seccomp.go]: Yeah. We could certainly try asking them to switch, but from their POV the `pkg-config` should technically be more correct and they probably aren't using bazel, I wouldn't bet on it... 🤔 

I'll let the other reviewers decide on the filtering or let that be another PR, I'd be happy to add it but I'm not sure if we should and it's not strictly necessary to get this fix 😅

**Comment by BenTheElder** [comment]
rebased

**Review by fejta — COMMENTED** [review]
/lgtm
/approve

**Comment by BenTheElder** [comment]
/assign @liggitt
@kubernetes/dep-approvers can't actually approve deps :( https://github.com/kubernetes/test-infra/issues/7690

**Review by cblecker — APPROVED** [review]
/lgtm
/approve

**Comment by liggitt** [comment]
/approve

for root go.{mod,sum}

**Comment by odinuge** [comment]
Is this something we should backport, ref. changelog for `v0.9.1` https://github.com/seccomp/libseccomp-golang/releases? We were running a version behind even `v0.9.0`.

Diff here:
https://github.com/seccomp/libseccomp-golang/compare/1b506fc7c24e...v0.9.1

**Comment by cblecker** [comment]
What would be the reason we'd backport?

**Pull request merged**
